source/server/server.cc

```diff
@@ -72,6 +72,7 @@ InstanceImpl::InstanceImpl(Options& options, Network::Address::InstanceConstShar
               e.what());
     thread_local_.shutdownGlobalThreading();
     thread_local_.shutdownThread();
+    stats_store_.shutdownThreading();
```

Comment:
This should happen in the same order as normal shutdown. Please check that.

Reply:
Updated. I'm almost thinking that I should extract the shutdown pattern in run() into its own method and have that used here and in the run() method.